src/CatchUp/CatchUp.php

```diff
@@ -140,6 +140,8 @@ public function run(EventStreamInterface $eventStream): void
         $iteration = 0;
         try {
             foreach ($eventStream->withMinimumSequenceNumber($highestAppliedSequenceNumber->next()) as $event) {
+                // WHY IS THIS CHECK NEEDED?
```

Comment:
ah the check is needed I guess because there is a short point in time where another projection could progress...

-> let's try to document this :)

Reply:
Exactly. But we might get around this with a slightly different catchup implementation I'd love to discuss with you sometimes next week!?

Reply:
Any news on this one?

Reply:
IIRC the alternative catchup implementation was based on a more coordinated process such that we can make sure that there is always ever one catch up per listener in progress.
But this comes with some other caveats and potential issues that we first have to address.

For now I'll document this (in the process of #10).

## Background:

The reason why this check is needed is because we release and re-acquire the lock in after one batch (see  https://github.com/neos/eventstore/blob/c6ebb743296d4ef3aeb0b9e0d5b056bd280051a8/src/CatchUp/CatchUp.php#L154-L155).
If multiple processes catch up the same listener this can lead to the same event being applied twice

Reply:
@bwaidelich how should we continue here - in light that this class will be moved to neos core - https://github.com/neos/eventstore/issues/18